biscuit-auth/src/token/authorizer.rs

```diff
@@ -31,10 +31,24 @@ pub struct Authorizer {
     pub(crate) blocks: Option<Vec<Block>>,
     pub(crate) public_key_to_block_id: HashMap<usize, Vec<usize>>,
     pub(crate) limits: AuthorizerLimits,
-    pub(crate) execution_time: Duration,
+    pub(crate) execution_time: Option<Duration>,
 }
 
 impl Authorizer {
+    pub fn run(&mut self) -> Result<Duration, error::Token> {
+        match self.execution_time {
+            Some(execution_time) => Ok(execution_time), // TODO should `run` be idempotent or raise an error if called twice?
```

Comment:
run being idempotent makes it really easy to lazily trigger evaluation in either authorize or query, so it is very important to have this ability. maybe as an internal function if we really want to make sure run triggers evaluation or fails if it can’t?

Reply:
Yep it's good that this is idempotent